src/signature_detect/bounding_box.py

```diff
@@ -0,0 +1,189 @@
+
+            if (h * w > self.min_region_size and h < copy_img.shape[0] and
+                    w < copy_img.shape[1]):
+
+                # cv2.rectangle(copy_img, (x, y), (x + w, y + h), (155, 155, 0), 1)
+                boxes.append([x, y, w, h])
+
+        np_boxes = np.array(boxes)
+        # sort the boxes by area size
+        area_size = list(map(lambda box: box[2] * box[3], np_boxes))
+        area_size = np.array(area_size)
+        area_dec_order = area_size.argsort()[::-1]
+        sorted_boxes = np_boxes[area_dec_order]
+
+        return sorted_boxes
+
+    def is_intersected(self, new_box, orignal_box) -> bool:
+        [x_a, y_a, w_a, h_a] = new_box
+        [x_b, y_b, w_b, h_b] = orignal_box
+
+        if y_a > y_b + h_b:
+            return False
+        if y_a + h_a < y_b:
+            return False
+        if x_a > x_b + w_b:
+            return False
+        if x_a + w_a < x_b:
+            return False
+        return True
+
+    def merge_boxes(self, box_a, box_b) -> list:
+        """
+        merge 2 intersected box into one
+        """
+        [x_a, y_a, w_a, h_a] = box_a
+        [x_b, y_b, w_b, h_b] = box_b
+
+        min_x = min(x_a, x_b)
+        min_y = min(y_a, y_b)
+        max_w = max(w_a, w_b, (x_b + w_b - x_a), (x_a + w_a - x_b))
+        max_h = max(h_a, h_b, (y_b + h_b - y_a), (y_a + h_a - y_b))
+
+        return [min_x, min_y, max_w, max_h]
+
+    def _remove_borders(self, box) -> Any:
+        """
+        remove the borders around the box
+        """
+        [x, y, w, h] = box
+        border = math.floor(min(w, h) * self.border_ratio)
+        return [x + border, y + border, w - border, h - border]
+
+    def boxes2regions(self, sorted_boxes) -> dict:
+        regions = {}
+
+        for box in sorted_boxes:
+            if len(regions) == 0:
+                regions[0] = box
+            else:
+                is_merged = False
+                for key, region in regions.items():
+                    if self.is_intersected(box, region) == True:
+                        new_region = self.merge_boxes(region, box)
+                        regions[key] = new_region
+                        is_merged = True
+                        break
+                if is_merged == False:
+                    key = len(regions)
+                    regions[key] = box
+        return regions
+
+    def bounding_boxes(self, mask, regions) -> dict:
+        results = {}
+        for key, region in regions.items():
+            # crop region
+            cropped_region = self._remove_borders(region)
+            [x, y, w, h] = cropped_region
+
+            image = Image.fromarray(mask)
+            cropped_image = image.crop((x, y, x + w, y + h))
+            cropped_mask = np.array(cropped_image)
+
+            results[key] = {
+                "cropped_region": cropped_region,
+                "cropped_mask": cropped_mask,
+            }
+
+        x = results[0]['cropped_region'][0]
```

Comment:
Ahhh! I get your point! Maybe we can create two public functions, one for returning cropped region, and the other returns cropped mask

Reply:
Yes!!! :)